### DISCUSSION TITLE
[Task Synchronization] - API / Webhook

### DISCUSSION THREAD

**Discussion by latetedemelon** [outline/outline#5677] [Ideas]
Hi Everyone,

It appears that there is quite an appetite out there for better handling of tasks.  One of the items I keep running into is that I have trouble finding / getting my tasks out of outline.  While ideally outline would be able to handle this, from the comments this might not be the direction the development team wants to go with?

To me, the easiest way to handle / approach this is to enhance the API to allow task creation / update events to trigger a webhook and at the same time make task updates available via the API.  So the workflow in question would be:

1. Work on outlining / writing and remember a task that is related or needs to be complete.
2. Create the task on the fly and move on.
3. Webhook fires this over to your task management software.  (Title and checked vs not)
4. Task is prioritized and worked on externally.  Syncing of task title happens if appropriate. 
5. Task is complete; outside software calls API to changed the checked status in outline.

Love the product @tommoor - I really apricate the hard work you put into this.

Related:
https://github.com/outline/outline/discussions/4845
https://github.com/outline/outline/discussions/3674
https://github.com/outline/outline/discussions/3370
https://github.com/outline/outline/discussions/1312

**Comment by tommoor** [comment]
Hi, what software are you currently using for task management?

**Reply by latetedemelon** [reply]
Hey Tom,

What don't I use (or more specifically, what haven't I tried) would be a better question.  If you're looking for a suggestion I would go with Vikunja: https://github.com/go-vikunja.

My reasoning it that's it's an open source, self-hostable app like yours, but doesn't appear to have any feature overlap with your product.    To me an integration would make both products stronger. 

Demo: https://try.vikunja.io/
OpenAPI Spec: https://try.vikunja.io/api/v1/docs
Webhooks are pending for next release: https://my.vikunja.cloud/tasks/558#share-auth-token=QFyzYEmEYfSyQfTOmIRSwLUpkFjboaBqQCnaPmWd

**Reply by latetedemelon** [reply]
@tommoor - I was experimenting with using the document / revision creation webhooks to get notified and trying to extract the tasks from the markdown.  It "works" but it only works for creation, it's not possible to keep the tasks in sync without an id to reference them as the text could change at any time.  

Anyway, hopeful this makes it on to the future bucket list :)

**Reply by StefanRiedAV** [reply]
I'd like to echo the pointer to vikunja. 
by far the most interesting open source modern task management, I came across.
Pro, 
you would imagine to include a Vikunja project on a specific Outline pages. 
You could already do this with the /embed construct in Outline.
Contra
The openID connector of vikunja is not open source. It is a paid add one.

**Reply by ucola** [reply]
@StefanRiedAV we setup vikunja an the openid with google is free on self hosted solutions...

**Reply by StefanRiedAV** [reply]
yes, you are right, I mixed this up with the hosted version. 
I have to learn how to write an native outline integration. There could be a vikunja integration which shows a certain project inline on the outline page.
... 
Let me know if anybody did something like this already.

**Reply by ucola** [reply]
@StefanRiedAV We would also be interested.

**Reply by malys** [reply]
At the end, is there a integration de vikunja?

**Reply by StefanRiedAV** [reply]
No. This integration has not been started. 
You can actually use a vikunja link as active embedded link in an outline page. 
This means instead of sync'ing task fto outline, you type immediately into vikunja.

**Reply by malys** [reply]
Thanks for your reply

**Reply by NeptunedGmbH** [reply]
I would also support an integration of vikunja or better ways to handle tasks. I wouldn't mind syncing tasks via API and n8n. currently it's a lot of manual labor to use outline and vikunja together. I would love to be able to create a new board from Outline and prefill it with tasks from outline
